### ISSUE TITLE
Include another YAML configuration in application.yml

### ISSUE THREAD

**Issue by temofey1989** [micronaut-projects/micronaut-core#10436]
### Feature description

In some cases, our main configuration (`applicaiton.yml`) is getting too big to manage.

It would be pretty great to have the possibility to include the content of some YAML into `application.yml`.
Quarkus has a similar feature. Example:
```yaml
...
quarkus:
  config:
    locations:
      - config-1.yml
      - config-2.yml
...
```

This feature can also be useful when you have a hierarchy of `applicaiton-*.yml` config files.
For instance, I have a `application-test.yml` which has a common configuration for the test.
```yaml
logger:
  levels:
    root: DEBUG
...
```
Then I can have `application-testdb.yml`:
```yaml
datasources:
  default:
     ...
```
Having possibility of including configuration from another file, I can just declare test like this:
```kotlin
@MicronautTest(environments = ["testdb"])
class MyTest : FreeSpec({
    ...
})
```
instead of:
```kotlin
@MicronautTest(environments = [Environment.TEST, "testdb"])
class MyTest : FreeSpec({
    ...
})
```

Thanks for the feedback.

**Comment by altro3** [comment]
@graemerocher  I agree that this feature is sorely missed. As correctly described in the ticket, this is NOT the same as profiles. Splitting one large file with settings into several small ones is just a decomposition of the configuration. But from the point of view of implementation, I think.you can do the same as quarkus, that is, include several profiles inside the configuration.

There is another way to use this feature: when you create several libraries inside which you want to set default settings and then to be able to connect a library with ready-made settings. Thus, you will need to register one line in the application.yml. But it's more of an idea.

**Comment by graemerocher** [comment]
You could automatically configure additional environment names in a context configurer that you place in a library or your app:

```java
@ContextConfigurer
public final class MyContextConfigurer implements ApplicationContextConfigurer {


    @Override
    public void configure(ApplicationContextBuilder builder) {
        builder.environments("testdb");
    }
}

```

Then that will always be activated and `application-testdb.yml`

**Comment by altro3** [comment]
Yes, it looks like what is needed. The only difference is that it is not possible to set this via the configuration in the yaml file.

Here is an example when this is required: for example, I have a set of zip codes for different countries - large yaml files. I want to load them in the prod, but be able to load them sometimes at local startup

If this is done through profiles, as you suggest, then I need to make a @ContextConfigurer that will add a zip-codes profile to the builder, depending on the selected profile or some system variable.

If it was possible to set this via a yaml file, I would just write something like this in application-local.yml:
```yaml
micronaut:
  environment:
    import: zip-codes
```
And to disable the loading of zip codes, I would just comment out this line, without editing the code.

I mean, I'm not sure if the current implementation is user-friendly, because it only works through code fixes.

**Comment by bpstelios10** [comment]
@graemerocher i m going to pick this one up

**Comment by graemerocher** [comment]
Ok 👍

**Comment by gonciarz** [comment]
This feature is very needed. I will give you another use case. I can't store credentials in source code. I want to keep them in a separate config file (e.g. application-secrets.yaml/properties outside of my source code.
Also I don't want to pass any argument to JVM or set environment variable. I know that it is doable with -Dmicronaut.config.files but I would like to avoid that.
Also this file could be set as optional.
Eventually I was able to configure custom context configurer and use property source loader and pass it to builder if an external file exist. However that's also a vector attack for production config. I don't want anyone to be able to include such file on prod. Though I manually try to detect env (system property or env variable) and build a logic based on that. This workaround is dirty and It would be great if native mechanism exist.

**Comment by dominictobias-bullish** [comment]
So is it currently not possible? 🤯

**Comment by dstepanov** [comment]
I'm going to add a way to include extra location in v5

**Comment by graemerocher** [comment]
PR Submitted
